tests/parse-module-name.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const parseModuleName = require('../lib/parse-module-name');
+
+describe('parse-module-name helper', function() {
+  it('should return applicaiton template when using a route local -component', function() {
```

Comment:
typo here, should read `application`
Also can you add a few more test for the null case?

```js
it('should return null for public components', function() {
  expect(parseModuleName('emberconf/src/ui/components/foo-bar/component-test.js').toEqual(null);
}
it('should return null for public components in test folder', function() {
  expect(parseModuleName('emberconf/tests/integration/components/foo-bar/component-test.js').toEqual(null);
}
```